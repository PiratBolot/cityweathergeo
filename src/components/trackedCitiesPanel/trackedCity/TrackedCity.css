.tracked_city {
    color: #738287;
    flex-grow: 1;
    background-color: white;
    margin: 5px;
    border-radius: 5px;
    padding: 10px 0px;
    box-shadow: 0 0 5px 0 #202020;
}
