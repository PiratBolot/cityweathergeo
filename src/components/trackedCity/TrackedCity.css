.tracked_city {
    color: #738287;
    flex-grow: 1;
    background-color: white;
    margin: 5px;
    border-radius: 5px;
    padding: 10px 0;
    box-shadow: 0 0 5px 0 #202020;
}

.wf_header {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding: 0 10px;
}

.city_name {
    font-size: 24px;
    font-weight: bold;
    padding: 0 5px;
}

.temperature_c {
    color: #435257;
    font-size: 40px;
    padding: 0 5px;
    line-height: 1;
}

.weather_icon {
    max-width: 40px;
    max-height: 40px;
    filter: drop-shadow(2px 4px 6px black);
}

.remove_city_button {
    display: inline-block;
    background-color: #435257;
    color: white;
    margin: 10px;
    padding: 10px;
    border: none;
    border-radius: 50%;
    box-shadow: 0 3px 20px rgba(0,0,0,.25);
}
