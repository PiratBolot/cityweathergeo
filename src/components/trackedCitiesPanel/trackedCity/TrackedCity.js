import React from "react";
import './TrackedCity.css';

import {getWeatherByCityName, parseWeatherResponse} from '../../../WeatherApi'
import WeatherProps from "../../weatherProps/WeatherProps";
import {connect} from "react-redux";
import {deleteTrackedCity, resetError, setErrorState} from "../../../actions/Actions";
import CityProps from "../../cityProps/CityProps";
import Loader from "../../preLoader/Loader";

class TrackedCity extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            city: props.city,
            loaded: false
        };
    }

    async componentDidMount() {
        let data = await getWeatherByCityName(this.props.city);
        if (data.status === "200") {
            if (this.props.trackedCities.indexOf(this.props.city) !== -1) {
                this.props.successAddCity();
            }
        } else {
            this.props.failureAddCity(data.status === "404" ? "Город " + this.props.city + " не найден" : "Ошибка API " + data.status);
            this.props.deleteCity(this.props.city);
        }
        let res = data.response;
        let parsedData = parseWeatherResponse(res);

        this.setState({loaded: !!res, data: res, parsedWeather: parsedData, city: res.name});
    }

    render = () => (
        <div className="weather_frame">
            <CityProps isCurrentCity={false} city={this.props.city} data={this.state.data} />
            {
                this.state.loaded ?
                    <WeatherProps parsedWeather={this.state.parsedWeather} />
                    :
                    <Loader />
            }
        </div>
    )
}

const mapStateToProps = (state) => ({
    trackedCities: state.trackedCities,
});

const mapDispatchToProps = (dispatch) => ({
    deleteCity: (city) => dispatch(deleteTrackedCity(city)),
    successAddCity: () => dispatch(resetError()),
    failureAddCity: (msg) => dispatch(setErrorState(msg))
});

export default connect(mapStateToProps, mapDispatchToProps)(TrackedCity);