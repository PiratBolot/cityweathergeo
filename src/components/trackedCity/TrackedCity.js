import React from "react";
import './TrackedCity.css';

import {getWeatherByCityName, parseWeatherResponse} from '../../WeatherApi'

import CircularSpinner from "../preLoader/CircularSpinner";
import {connect} from "react-redux";
import {deleteTrackedCity} from "../../actions/Actions";

class TrackedCity extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            city: props.city,
            loaded: false
        };

        this.removeTrackedCity = this.removeTrackedCity.bind(this);
    }

    async componentDidMount() {
        let data = await getWeatherByCityName(this.props.city);
        let res = data.response;
        let parsedData = parseWeatherResponse(res);

        /* Uncomment if you want to see the preLoader explicitly
        var answer = prompt('question', 'defaultAnswer');
        */

        this.setState({loaded: !!res, data: res, parsedWeather: parsedData, city: res.name});
    }

    removeTrackedCity = async (e) => {
        e.preventDefault();
        this.props.deleteFavCity(this.state.city);
    };

    render = () => (
        <div className="weather_frame">
            <div className="wf_header">
                <div className="city_name">
                    {this.state.city}
                </div>
                {
                    this.state.data ?
                        <div className="temperature_c">
                            <span>{(this.state.data.main.temp - 273.14).toFixed(1) + "°C"}</span>
                        </div> : ""
                }
                {
                    this.state.data ?
                        <img className="weather_icon" src={"http://openweathermap.org/img/wn/" +
                        this.state.data.weather[0].icon + ".png"} alt=""
                        /> : ""
                }
                <button className="remove_city_button" onClick={this.removeTrackedCity}>X</button>
            </div>
            {
                this.state.loaded ?
                    <div className="wn_body">
                        {
                            this.state.parsedWeather.map((e) =>
                                <div key={e.key} className="wn_line">
                                    <div className="info_label">{e.key}</div>
                                    <div className="info_value">{e.value}</div>
                                </div>
                            )
                        }
                    </div>
                    :
                    <div className="image_container_overlay">
                        <p className="label">Подождите, данные загружаются</p>
                        <CircularSpinner />
                    </div>
            }
        </div>
    )
}

const mapStateToProps = (state) => ({});

const mapDispatchToProps = (dispatch) => ({
    deleteFavCity: (city) => dispatch(deleteTrackedCity(city))
});

export default connect(mapStateToProps, mapDispatchToProps)(TrackedCity);