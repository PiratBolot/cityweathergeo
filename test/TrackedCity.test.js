import React from 'react';
import {shallow} from '../enzyme/enzyme';
import {applyMiddleware, createStore} from "redux";
import rootReducer from "../src/reducers/Reducers";
import thunk from "redux-thunk";
import toJson from "enzyme-to-json";
import TrackedCity from "../src/components/trackedCitiesPanel/trackedCity/TrackedCity";

describe('Tracked City', () => {
    const city = "Syktyvkar";
    const initialState = {
        trackedCities: ['Syktyvkar', 'Moscow', 'Kiev'],
        errorMessage: {isError: false, errorMsg: ""},
        geolocation: {}
    };
    let store,wrapper;
    beforeEach(()=>{
        store = createStore(rootReducer, initialState, applyMiddleware(thunk));
        wrapper = shallow(<TrackedCity city={city} store={store} />);
    });
    it('+++capturing Snapshot of TrackedCity', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});