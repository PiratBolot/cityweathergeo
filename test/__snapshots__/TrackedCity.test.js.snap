// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracked City +++capturing Snapshot of TrackedCity 1`] = `
<ContextProvider
  value={null}
>
  <TrackedCity
    city="Syktyvkar"
    deleteCity={[Function]}
    failureAddCity={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    successAddCity={[Function]}
    trackedCities={
      Array [
        "Syktyvkar",
        "Moscow",
        "Kiev",
      ]
    }
  />
</ContextProvider>
`;
